griptape/tools/web_search/tool.py

```diff
@@ -15,25 +15,30 @@
 
 @define
 class WebSearch(BaseTool):
-    web_search_driver: BaseWebSearchDriver = field(default=None, kw_only=True)
+    web_search_driver: BaseWebSearchDriver = field(kw_only=True)
 
     @activity(
         config={
-            "description": "Can be used for searching the web",
+            "description": "Can be used for searching the web via the {{ _self.web_search_driver.__class__.__name__}}.",
             "schema": Schema(
                 {
                     Literal(
                         "query",
                         description="Search engine request that returns a list of pages with titles, descriptions, and URLs",
                     ): str,
+                    Optional(
+                        "params",
+                        description="Additional params that can be passed with the request.",
```

Comment:
Not sure if it would help with the issues you were having, but maybe you could elaborate a little more on examples of what "params" is used for?

Reply:
Basically anything that you might want the LLM to pass at runtime. The date range example is what prompted this change, but in theory you could prompt the LLM to use any [of these](https://developers.google.com/custom-search/v1/reference/rest/v1/cse/list).

Reply:
Oh, I didn't mean elaborate for me, I meant for the LLM, by making the description less generic